contrib/scripts/unloaddb.sh

```diff
@@ -0,0 +1,402 @@
+                        let "num_tables_in_slot++"
+                        echo "${table_selected[i]}" >> $file
+                fi
+        done
+
+        if [ $verbose = "yes" ];then
+                echo "Proc $slot_num: num tables: $num_tables_in_slot, ${slot_size[$slot_num]} bytes"
+        fi
+
+        echo "process $slot_num: starting" > unloaddb_$slot_num.status
+
+	cubrid unloaddb $user $pass --input-class-only --input-class-file $file $database &
+        pid=$!
+
+        echo $pid > unloaddb_$slot_num.pid
+
+        wait $pid
+
+        if [ $? -ne 0 ];then
+                msg="Failed"
+                echo "process $slot_num: failed" > unloaddb_$slot_num.status
+        else
+                echo "process $slot_num: success" > unloaddb_$slot_num.status
+        fi
+
+        if [ $verbose = "yes" ];then
+                echo "Finished Proc $slot_num ($msg)"
+        fi
+}
+
+function analyze_table_info ()
+{
+        local found=0
+        local line
+        local idx=0
+        local table_name
+        local db=$database
+
+        # Get all table names from CATALOG
+
+        if [ X$from_file != X"" ];then  # Read table name from file
+                result=$(cat $from_file)
+        else
+                result=$(csql $user $pass -c "select class_name from db_class where is_system_class = 'NO' AND class_type = 'CLASS' order by class_name" $db)
+        fi
+
+        for token in $result
+        do
+                if  [ X$from_file = X"" ] && [ $found -eq 0 ];then
+                        str=$(echo $token | grep "====") # skip until we found table names in csql
+                        if [ $? -eq 0 ];then
+                                found=1
+                        fi
+                        continue
+                fi
+
+                if [ X$from_file = X"" ] && [ ${token:0:1} != "'" ];then
+                        break
+                fi
+
+                table_selected[idx]="$token"
+
+                if [ X$from_file = X"" ];then        # remove single quota from csql output
+                        table_selected[idx]=$(echo ${table_selected[idx]} | sed "s/[\']//g")
+                fi
+
+                let "idx++"
+
+        done
+
+        # Get the size of all tables
+        n=1
+        for ((i = 0; i < ${#table_selected[@]}; i++))
+        do
+                table_name=${table_selected[i]}
+                this_table_size=$(get_table_size $table_name)
+
+                if [ -z $this_table_size ];then
+                        echo "Unknown table: $table_name"
+                        exit
+                fi
+
+                table_size[$num_tables]="$this_table_size"
+                let "num_tables++"
+        done
+}
+
+# MAIN
+#
+trap "cleanup" SIGHUP SIGINT SIGTERM
+set -o monitor
+
+extract_db_name $*
+get_options "$@"
+
+if [ ! -d $target_dir ];then
+        echo "$target_dir: directory not exists or permission denied"
+        exit
+else
+   silent_cd $target_dir
+fi
+
```

Comment:
Instead of checking whether the cub_server of database is running in verify_user_pass,
How about check it here explicitly ?

Reply:
I agree, it is moved from verify_user_pass () to MAIN.
Thank you for your comment!